proxy/start_container_interceptor.go

```diff
@@ -21,7 +21,11 @@ func (i *startContainerInterceptor) InterceptResponse(r *http.Response) error {
 		return err
 	}
 
-	cidrs, ok := i.proxy.weaveCIDRsFromConfig(container.Config)
+	if !validNetworkMode(container.HostConfig) {
+		Debug.Printf("Ignoring container %s with --net=%s", container.ID, networkMode(container.HostConfig))
+		return nil
+	}
```

Comment:
So now we are checking twice, both here and in `weaveCIDRsFromConfig`. Not great.

I suggest changing the `ok` return of `weaveCIDRsFromConfig` to a messsage (or error?) instead, which we can then log.